### PR TITLE
travisci and styleci configuration upgrade

### PR DESCRIPTION
**Pull request by Glavic**
Before we continue resolving all the issues and PR's, travisci and styleci should return successful builds. Based on that all this files were modified, which don't break anything.

### CODE REVIEW

**Comment by briannesbitt** [comment]
Does changing the function type hints from Carbon to self affect code editors offering type hints?  I seem to remember this being done and then reverted for some reason.  (Maybe I am thinking of the return types)

**Comment by kylekatarnls** [comment]
This would mean IDE bug. With eq(self $dt) PHPStorm says me eq needs a Carbon instance.

If someone complaints his IDE does not support, I would suggest to report an issue to the IDE owner.

**Comment by Glavic** [comment]
@briannesbitt: I cannot find and history in `src/Carbon/Carbon.php` that contains `self $dt`, so this was probably never implemented?

I have tried PHPstorm and vscode; all autocomplete are working...

phpstorm

vscode

**Comment by kylekatarnls** [comment]
vscode and PHPStorm have the same autocomplete engine so it's not so objective XD

**Comment by kylekatarnls** [comment]
But even if it would fail, it should not be taken into account. IDE does not support all magic methods properly but it's not a reason to avoid them. IDE plugins have to evolve according to new features not the opposite.

**Comment by briannesbitt** [comment]
Ya, looks like I was thinking of the return value.  (static|Carbon|self) and working nicely with phpdoc.  I think I am good to go here.

File: src/Carbon/Lang/bg.php [hunk 1]
```diff
@@ -21,7 +21,7 @@
     'hour' => '1 час|:count часа',
     'h' => '1 час|:count часа',
     'minute' => '1 минута|:count минути',
-    'm' => '1 минута|:count минути',
+    'min' => '1 минута|:count минути',
```
[the remark this answers is not in the document]
Comment (Glavic) [reply src/Carbon/Lang/bg.php]: Because I also wanted to solve phpcs errors.

**Comment by Ovsyanka** [comment]
Why it is not merged yet? When it will be? Just want to know because I waiting to rebase my PRs.

**Comment by kylekatarnls** [comment]
Hi @Ovsyanka this PR is only 2-hours old and it's a big one. Double-checking it will avoid you to rebase multiple times ;)

**Pull request merged**